### PR TITLE
Vulnerability section part2

### PR DESCRIPTION
**Pull request by zpevma**
Another 4 elements of Vulnerability section. Last one missing is `CVSSScoreSets`, the tricky one.

### CODE REVIEW

File: cvrf2csaf/section_handlers/vulnerability.py [hunk 1]
```diff
@@ -14,6 +15,76 @@ def __init__(self, config):
...
+
+            threats.append(threat)
+
+        return threats
+
+    @staticmethod
+    def _handle_remediations(root_element):
+        remediations = []
+        for remediation_elem in root_element.Remediation:
+            remediation = {'category': remediation_elem.attrib['Type'].lower().replace(' ', '_'),
```
Comment (tschmidtb51) [inline cvrf2csaf/section_handlers/vulnerability.py:66]: `Will Not Fix` was renamed to `no_fix_planned`.

**Review by tschmidtb51 — COMMENTED** [review]
Minor issue that was not documented clearly before => conversion of `Will Not Fix` to `no_fix_planned`.

File: cvrf2csaf/section_handlers/vulnerability.py [hunk 2]
```diff
@@ -14,6 +15,76 @@ def __init__(self, config):
     def _process_mandatory_elements(self, root_element):
         pass
 
+    @staticmethod
+    def _handle_involvements(root_element):
+        # TODO: potential handling of missing `date` entry
```
[the remark this answers is not in the document]
Comment (cgi1) [reply cvrf2csaf/section_handlers/vulnerability.py:20]: @zpevma: Please check where this todo comes from.

File: cvrf2csaf/section_handlers/vulnerability.py (hunk 1 above)
Comment (cgi1) [reply cvrf2csaf/section_handlers/vulnerability.py:66]: please fix no_fix_planned.

**Comment by cgi1** [comment]
Consulted about the comments with @tschmidtb51 - Please check comment ToDo and minor fixes. If comment ToDo is being removed, PR is approved and can be merged without further review.

File: cvrf2csaf/section_handlers/vulnerability.py [hunk 3]
```diff
@@ -42,12 +113,18 @@ def _process_optional_elements(self, root_element):
             vulnerability['id'] = {'system_name': root_element.ID.attrib['SystemName'],
                                    'text': root_element.ID.text}
 
+        if hasattr(root_element, 'Involvements'):
+            vulnerability['involvements'] = self._handle_involvements(root_element.Involvements)
+
         if hasattr(root_element, 'Notes'):
             # reuse DocumentNotes handler
             notes_handler = DocumentNotes()
```
Comment (sthagen) [inline cvrf2csaf/section_handlers/vulnerability.py:121]: (Was not introduced in thi change set) Or, adapt the architecture to provide a `Notes()` (and as seen further below) a `References()` handler to then mix these in where needed?

File: cvrf2csaf/section_handlers/vulnerability.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (zpevma) [reply cvrf2csaf/section_handlers/vulnerability.py:20]: Indeed, `date` is not part of CVRF 1.2. No TODO here is needed.

**Review by tschmidtb51 — APPROVED** [review]

File: cvrf2csaf/section_handlers/vulnerability.py (hunk 3 above)
Comment (tschmidtb51) [reply cvrf2csaf/section_handlers/vulnerability.py:127]: Agree. That should be done in refactoring...

Comment (zpevma) [reply cvrf2csaf/section_handlers/vulnerability.py:127]: Will do in another PR.

**Pull request merged**